code/iomethods.py

```diff
@@ -226,132 +247,123 @@ def create_main_directory(mainDirStr):
 
-        # Get the lat/lon info data (different resolution)
-        latminNETCDF = utils.find_nearest(alllatsraw, float(userVariables.LATMIN))
-        latmaxNETCDF = utils.find_nearest(alllatsraw, float(userVariables.LATMAX))
-        lonminNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMIN))
-        lonmaxNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMAX))
-        latminIndex = (np.where(alllatsraw == latminNETCDF))[0][0]
-        latmaxIndex = (np.where(alllatsraw == latmaxNETCDF))[0][0]
-        lonminIndex = (np.where(alllonsraw == lonminNETCDF))[0][0]
-        lonmaxIndex = (np.where(alllonsraw == lonmaxNETCDF))[0][0]
+        tmp.close()
+    elif fileType == 'binary':
+        alllatsraw, alllonsraw, _ = read_MERG_pixel_file(userVariables.filelist[0])
 
-        # Subsetting the data
-        latsraw = alllatsraw[latminIndex: latmaxIndex]
-        lonsraw = alllonsraw[lonminIndex:lonmaxIndex]
+    # Get the lat/lon info data (different resolution)
+    latminNETCDF = utils.find_nearest(alllatsraw, float(userVariables.LATMIN))
+    latmaxNETCDF = utils.find_nearest(alllatsraw, float(userVariables.LATMAX))
+    lonminNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMIN))
+    lonmaxNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMAX))
+    latminIndex = (np.where(alllatsraw == latminNETCDF))[0][0]
+    latmaxIndex = (np.where(alllatsraw == latmaxNETCDF))[0][0]
+    lonminIndex = (np.where(alllonsraw == lonminNETCDF))[0][0]
+    lonmaxIndex = (np.where(alllonsraw == lonmaxNETCDF))[0][0]
 
-        LON, LAT = np.meshgrid(lonsraw, latsraw)
+    # Subsetting the data
+    latsraw = alllatsraw[latminIndex:latmaxIndex]
+    lonsraw = alllonsraw[lonminIndex:lonmaxIndex]
 
-        latsraw = []
-        lonsraw = []
-        tmp.close()
+    LON, LAT = np.meshgrid(lonsraw, latsraw)
 
-    for files in userVariables.filelist:
-        try:
-            thisFile = Dataset(files, 'r', format='NETCDF4')
-            # Clip the dataset according to user lat, lon coordinates
-            # Mask the data and fill with zeros for later
-            tempRaw = thisFile.variables[varName][:, latminIndex:latmaxIndex, lonminIndex:lonmaxIndex].astype('int16')
-            tempMask = ma.masked_array(tempRaw, mask=(tempRaw > userVariables.T_BB_MAX), fill_value=0)
-            # Get the actual values that the mask returned
+    timeName = 'time'
+    for file in userVariables.filelist:
+        if fileType == 'netCDF':
+            try:
+                thisFile = netCDF4.Dataset(file, 'r', format='NETCDF4')
+
+                # Clip the dataset according to user lat, lon coordinates
+                # Mask the data and fill with zeros for later
+                tempRaw = thisFile.variables[varName][:, latminIndex:latmaxIndex, lonminIndex:lonmaxIndex].astype('int16')
+                tempMask = ma.masked_array(tempRaw, mask=(tempRaw > userVariables.T_BB_MAX), fill_value=0)
```

Comment:
A fill_value of 0 ma not be wise because given the data associated with the data, it may be counted as a valid value. For this particular variable (temperature in Kelvin and most vars in meteorological applications) a fill_value of -999 is better